examples/platform/policy_host_based_firewall.py

```diff
@@ -0,0 +1,152 @@
+        print("Rules:")
+        # the command line argument of "--all_details" is used here
+        if args.all_details:
+            for r in p.rules:
+                print("  {0}: {1} when {2} {3} is {4}".format(r.get("id"), r.get("action"),
+                                                              r.get("application", {}).get("type"),
+                                                              r.get("application", {}).get("value"),
+                                                              r.get("operation")))
+            print("Core Prevention")
+            if p.core_prevention_rule_configs is None:
+                print("No Core Prevention Rules")
+            else:
+                if args.all_details:
+                    for cprc in p.core_prevention_rule_configs.values():
+                        print(" {0}".format(cprc.name))
+        else:
+            print("Details not requested.")
+
+        print("Host-Based Firewall")
+        if p.host_based_firewall_rule_config is None:
+            print("No Host-Based Firewall Rules")
+        else:
+            if args.all_details:
+                hbfwr = p.host_based_firewall_rule_config
+                print("Rule config name: {0}".format(hbfwr.name))
+                if hbfwr.rule_groups is None:
+                    print("No rule groups")
+                else:
+                    for rg in hbfwr.rule_groups:
+                        print("Rule Group Name: {0}".format(rg.name))
+                        if rg.rules is None:
+                            print("No rules")
+                        else:
+                            for r in rg.rules:
+                                print("rule: {0}".format(r))
+            else:
+                print("Details not requested.")
+
+        print("")
+        print("End of Policy Object")
+        print("")
+
+
+def add_hbfw_rule(cb, args):
+    """Create a rule and add it to the rule group specified in the input args"""
+    # get the policy specified in the input parameters
+    policy = cb.select(Policy, args.policy)
+    print("id: {0}.  name: {1}".format(policy.id, policy.name))
+    # If the policy does not yet have a host-based firewall section then create it
+    if policy.host_based_firewall_rule_config is None:
+        policy.host_based_firewall_rule_config = HostBasedFirewallRuleConfig(cb, policy)
+
+    # get the rule config in a variable to work with
+    rc = policy.host_based_firewall_rule_config
+    # create a new rule group
+    rg = rc.new_rule_group(args.rule_group_name, args.rule_group_desc)
+    # add the rule group to the rule config
+    rc.append_rule_group(rg)  # to do remove this
+
+    # create two rules
+    # name = SDK Example Rule One, action = ALLOW, direction = IN, protocol = TCP, remote ip address = 15.16.17.18
+    r1 = rc.new_rule("SDK Example Rule One", "ALLOW", "IN", "TCP", "15.16.17.18")
```

Comment:
Do we want to utilize pythons `input()` to ask for this information as it would be great to make our examples a little more useful

Reply:
Changed to use this.